src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -552,97 +540,86 @@ private static MoveValidationResult validateBasic(final GameData data,
       return result.setErrorReturnResult("Transports not found in route end");
     }
     if (!isEditMode) {
-      // make sure all units are at least friendly
+
+      // Make sure all units are at least friendly
       for (final Unit unit : CollectionUtils.getMatches(units, Matches.enemyUnit(player, data))) {
         result.addDisallowedUnit("Can only move friendly units", unit);
       }
-      // check we have enough movement
-      // exclude transported units
-      final Collection<Unit> moveTest;
+
+      // Ensure all air transports are included
+      for (Unit airTransport : newDependents.keySet()) {
+        if (!units.contains(airTransport)) {
+          for (Unit unit : newDependents.get(airTransport)) {
+            if (units.contains(unit)) {
+              result.addDisallowedUnit("Not all units have enough movement", unit);
+            }
+          }
+        }
+      }
+
+      // Ignore transported units
+      Collection<Unit> moveTest = new ArrayList<>(units);
       if (route.getStart().isWater()) {
         moveTest = MoveValidator.getNonLand(units);
-      } else {
-        moveTest = units;
       }
+      final Map<Unit, Collection<Unit>> dependentsMap =
+          getDependents(CollectionUtils.getMatches(units, Matches.unitCanTransport()));
```

Comment:
I'm noting that we call `getDependents` to get some data and then the next few lines transform that data into a more manageable form. It seems that the API on `getDependents` is not quite what we want, hence the need to transform the API. That basically means we should not call `getDepedents` for this purpose. Two options I see here are to in-line the code for getDependents (not very desirable as this would be duplication), or to add another helper method that would return a more usable dataset and remove some of the complexity here.

Reply:
getDependents is used in multiple places so can't easily change its return type. The transformation of the return type is pretty minimal and mostly specific to this method as its combining the existing dependents with new dependents into a single set. I don't think a helper method here would be that useful as its only 3 lines currently.

Reply:
A very meaty 3 lines! I'm actually still not quite sure what the transform does, `dependent` is not well defined yet in my mind for this context.

`getDependents` is itself only 4 lines. It only does 50% of what needs to be done here, just saying : )

This method is long enough that removing any complexity I think is a win. It is really dense, one could spend a good hour or two on just these few dozen lines of code (that is a bad thing, we have around 200k lines of code)